asf_public_discourse_home_decarbonisation/analysis/FAQ_analysis/BERTopic_first_analysis.py

```diff
@@ -0,0 +1,169 @@
+    Generates and saves a visualisation of topics identified by the BERTopic model.
+
+    Args:
+        topic_model (BERTopic): The BERTopic model after fitting to data.
+    """
+    fig = topic_model.visualize_topics()
+    fig.write_image(figure_file_path + "topic_visualisation.png")
+
+
+def visualise_barchart(topic_model: BERTopic, figure_file_path: str) -> None:
+    """
+    Generates and saves a barchart visualisation of the top n topics identified by the BERTopic model.
+
+    Args:
+        topic_model (BERTopic): The BERTopic model after fitting to data.
+    """
+    fig_barchart = topic_model.visualize_barchart(top_n_topics=16, n_words=10)
+    fig_barchart.write_image(figure_file_path + "topic_visualisation_barchart.png")
+
+
+def visualise_hierarchy(topic_model: BERTopic, figure_file_path: str) -> None:
+    """
+    Generates and saves a hierarchical visualisation of topics identified by the BERTopic model.
+
+    Args:
+        topic_model (BERTopic): The BERTopic model after fitting to data.
+    """
+    fig_hierarchy = topic_model.visualize_hierarchy()
+    fig_hierarchy.write_image(figure_file_path + "topic_visualisation_hierarchy.png")
+
+
+def plot_topic_distribution(topic_model: BERTopic, figure_file_path: str) -> None:
+    """
+    Plots and saves the distribution of the top topics identified by the BERTopic model.
+
+    Args:
+        topic_model (BERTopic): The BERTopic model after fitting to data.
+    """
+    topic_counts = topic_model.get_topic_info()["Count"][1:17]
+    topic_labels = topic_model.get_topic_info()["Name"][1:17].str.replace("_", " ")
+    plt.figure(figsize=(14, 8))
+    plt.barh(topic_labels, topic_counts, color=NESTA_COLOURS[0])
+    plt.ylabel("Topics")
+    plt.xlabel("Count")
+    plt.title("Topic Distribution")
+    plt.tight_layout()
+    plt.savefig(
+        figure_file_path + "topic_distribution.png", dpi=300, bbox_inches="tight"
+    )
+
+
+def main():
+    """
+    Main function to execute the topic modeling workflow.
+
+    This function sets up the plotting styles, loads the data, creates a topic model from the questions,
+    and generates visualisations for topics, barchart, hierarchy, and topic distribution.
+    """
+    set_plotting_styles()
+    args = create_argparser()
+    category = args.category
+    forum = args.forum
+    input_data = os.path.join(
+        PROJECT_DIR,
+        f"outputs/data/extracted_questions/{forum}/forum_{category}/extracted_questions_{category}_all.csv",
+    )
+    figure_path = os.path.join(
+        PROJECT_DIR, f"outputs/figures/extracted_questions/{forum}/forum_{category}/"
+    )
+    os.makedirs(figure_path, exist_ok=True)
+    questions = load_data(input_data)
+    topic_model, topics, probabilities = create_topic_model(questions)
+    visualise_topics(topic_model, figure_path)
+    visualise_barchart(topic_model, figure_path)
+    visualise_hierarchy(topic_model, figure_path)
+    plot_topic_distribution(topic_model, figure_path)
+
+
```

Comment:
A few more things that are worth checking/saving (I actually have a function for these in my PR, but for now you can have the function calls below). `docs_info` or whatever you decide to call it will tell you info such as:
- which cluster a specific question belongs to;
- if a question is representative of the cluster or not;
- etc

```suggestion
    plot_topic_distribution(topic_model, figure_path)
    
    doc_info = topic_model.get_document_info(questions)
    
    # save this df
    #...
    
    topics_info = topic_model.get_topic_info()
    topics_info["%"] = topics_info["Count"] / len(questions) * 100
    # save this df
    #...


```

Reply:
Ah nice! Thanks for these suggestions 👍